drafts/bitcartel-payment-disclosure/draft.rst

```diff
@@ -0,0 +1,345 @@
+
+Using the JoinSplitSig key pair in this way also means that a sender can make payments without having to first provide a refund address or other details at the time of the original payment.  If at a later date the sender and recipient need to confirm the payment, the sender can sign and send any data required to the recipient, who can in turn verify it.
+
+
+Known Issues
+------------
+Both plaintext outputs of a joinsplit are currently encrypted with symmetric keys derived from the same ephemeral secret key, as discusshed here: https://github.com/zcash/zcash/issues/558#issuecomment-167819936
+
+This means that a payment disclosure which includes the ephemeral secret key intended to decipher the note ciphertext belonging to a particular joinsplit output, creates a privacy leak by also deciphering the other joinsplit output and revealing it's amount and payment address.
+
+Note the current implementation of the ``z_sendmany`` RPC call:
+
+- When sending from a transparent address to shielded addresses, each shielded address will occupy one output of a joinsplit.  Thus a payment address to a different recipient could be revealed.
+
+- When sending involves chained joinsplits, for each joinsplit, one output is for payment to a recipient's address, while the other output is used as change back to the sender.  Thus the change address will be revealed, which is the sender's own shielded address.
+
+A proposal to prevent information leakage of change addresses is currently under development [KDFT].  TODO: Does this block ZIP approval?
+
+Also note that an independent third party cannot know for sure if the sender and recipient are not colluding to hide value transfer.  JoinSplits have two inputs and two outputs.  The recipient may identify their receiving address for one output, but not disclose the fact that the other output also belongs to them.
+
+When a shielded transaction is created sucessfully and accepted into the local mempool with ``z_sendmany`` or ``z_shieldcoinbase``, the payment disclosure database is updated with information necessary to create a payment disclosure.  However, the transaction itself may never be mined and confirmed in the blockchain, rendering the database entry itself redundant and available for purging at a later date e.g. garbage collection. 
+
+.. [KDFT] https://github.com/zcash/zcash/issues/2102
+
+
+Specification
+-------------
+When creating a shielded transaction, for each JoinSplit output, a data structure is created to record the following fields:
+
+- transaction id
+- index [0..len-1] of JoinSplit in array of JoinSplits contained in transaction
+- index [0..1] of JoinSplit output
+- recipient's payment address is a shielded address ``(a_pk, pk_enc)`` [PROTOCOL] §3.1 Payment Addresses and Keys
+- symmetric key used to encrypt the note, also referred to as the ephemeral secret key ``esk`` [PROTOCOL] §4.10.1 Generate a new KA (public, private) key pair ``(epk, esk)``.
```

Comment:
``esk`` is not a symmetric key, it's a private key. In general avoid saying "ephemeral secret key" because that's confusing. Instead say:

* ephemeral private key ``esk`` used to encrypt the note

Reply:
ack